ml-dhs-aws/src/main/java/com/marklogic/client/dhs/JobRunner.java

```diff
@@ -64,22 +87,64 @@ public void run(DatabaseClient client, InputStream csvRecords, InputStream jobCo
                         throwable.printStackTrace(System.out);
                      });
 
-         ObjectMapper objectMapper = new ObjectMapper();
-         ObjectNode jobControlObj = (ObjectNode) objectMapper.readTree(jobControl);
-         ObjectNode jobDef        = (ObjectNode) jobControlObj.get("job");
-         ObjectNode recordDef     = (ObjectNode) jobControlObj.get("record");
 
          ObjectLoader loader = new ObjectLoader(batcher, recordDef, getJobDirectory(), documentMetadata);
-         CSVConverter converter = new CSVConverter();
-
-// TODO: write before job document in /jobStart and /jobs/ID collections or send before job payload to DHF endpoint
+         CSVConverter converter = new CSVConverter();  
 
+         Iterator<ObjectNode> itr = converter.convertObject(csvRecords);
+         if(!itr.hasNext()) {
+        	 throw new MarkLogicIOException("No header found.");
+         }
+         ObjectNode csvNode = itr.next();
+         
+         String headerValue = csvNode.fieldNames().toString();
+         
+   // write before job document in /jobStart and /jobs/ID collections or send before job payload to DHF endpoint        
+         String beforeDocId = "/jobs/"+id+"/beforeJob.json";
+         
+         DocumentMetadataHandle beforeDocumentMetadata = new DocumentMetadataHandle();
+         beforeDocumentMetadata.withCollections("/jobs/"+beforeDocId, "/beforeJob");
+         
+         String ingestionStartTime = LocalDateTime.now().toString();
+         
+         ObjectNode beforeDocRoot = objectMapper.createObjectNode();
+         beforeDocRoot.put("job id", id);
```

Comment:
Both here and in the after document, it might be better to use the camelCase jobId instead of the space-separated job id as the name of the key.  Keys with embedded spaces require bracket syntax in JavaScript but keys that resemble variable names can use period syntax in JavaScript.